### PR TITLE
[Function](ELT) Add elt function for doris.

### LINKED ISSUE #11990: [Enhancement] Add elt function for doris.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Returns the string at specified index. Returns NULL if there is no string at specified index.

Syntax:
`VARCHAR elt (INT,VARCHAR,...)`

Example:
mysql> select elt(1, 'aaa', 'bbb');
+----------------------+
 elt(1, 'aaa', 'bbb')         
+----------------------+
 aaa                                 
+----------------------+
mysql> select elt(2, 'aaa', 'bbb');
+-----------------------+
 elt(2, 'aaa', 'bbb')          
+-----------------------+
 bbb                                 
+-----------------------+
mysql> select elt(0, 'aaa', 'bbb');
+-----------------------+
 elt(0, 'aaa', 'bbb')          
+-----------------------+
 NULL                              
+-----------------------+
mysql> select elt(3, 'aaa', 'bbb');
+-----------------------+
 elt(3, 'aaa', 'bbb')          
+-----------------------+
 NULL                               
+-----------------------+

### Solution

Add elt function for doris.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yongjinhou**
# Proposed changes

Issue Number: close #11990

## Problem summary

Add elt function for doris.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/spill_sort_node.cc [hunk 1]
```diff
@@ -96,20 +96,20 @@ Status SpillSortNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* e
     DCHECK_EQ(row_batch->num_rows(), 0);
     RETURN_IF_ERROR(_sorter->get_next(row_batch, eos));
     while ((_num_rows_skipped < _offset)) {
-        _num_rows_skipped += row_batch->num_rows();
+        _num_rows_skipped += row_batch->num_rows();
```
Comment (Gabriel39) [inline be/src/exec/spill_sort_node.cc:99]:
```suggestion
        _num_rows_skipped += row_batch->num_rows();
```

File: be/src/exec/spill_sort_node.cc [hunk 2]
```diff
@@ -96,20 +96,20 @@ Status SpillSortNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* e
...
         } else {
-            row_batch->set_num_rows(0);
+            row_batch->set_num_rows(0); 
         }
         if (rows_to_keep > 0 || *eos) {
             break;
         }
         RETURN_IF_ERROR(_sorter->get_next(row_batch, eos));
-    }
+    }
```
Comment (Gabriel39) [inline be/src/exec/spill_sort_node.cc:112]:
```suggestion
    }
```

File: be/src/exec/spill_sort_node.cc [hunk 3]
```diff
@@ -96,20 +96,20 @@ Status SpillSortNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* e
...
-        _num_rows_skipped += row_batch->num_rows();
+        _num_rows_skipped += row_batch->num_rows(); 
         // Throw away rows in the output batch until the offset is skipped.
         int rows_to_keep = _num_rows_skipped - _offset;
         if (rows_to_keep > 0) {
             row_batch->copy_rows(0, row_batch->num_rows() - rows_to_keep, rows_to_keep);
             row_batch->set_num_rows(rows_to_keep);
         } else {
-            row_batch->set_num_rows(0);
+            row_batch->set_num_rows(0);
```
Comment (Gabriel39) [inline be/src/exec/spill_sort_node.cc:106]:
```suggestion
            row_batch->set_num_rows(0);
```

File: be/src/vec/functions/function_string.h [hunk 4]
```diff
@@ -491,6 +491,51 @@ class FunctionStringConcat : public IFunction {
...
+class FunctionStringElt : public IFunction {
+public:
+    static constexpr auto name = "elt";
+    static FunctionPtr create() { return std::make_shared<FunctionStringElt>(); }
+    String get_name() const override { return name; }
+    size_t get_number_of_arguments() const override { return 0; }
+    bool is_variadic() const override { return true; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
```
Comment (Gabriel39) [inline be/src/vec/functions/function_string.h:503]: Should be `Nullable(String)` here

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Comment by yiguolei** [comment]
1. Could you please add some SQL Based regression test?

**Pull request closed without merging**